test/test_array.cpp

```diff
@@ -72,6 +72,121 @@ namespace sparrow
             }
         }
         TEST_CASE_TEMPLATE_APPLY(access_operator_id, testing_types);
+
+        TEST_CASE_TEMPLATE_DEFINE("owns_arrow_strucure", AR, owns_arrow_structure_id)
+        {
+            SUBCASE("owning")
+            {
+                constexpr size_t size = 10;
+                using scalar_value_type = typename AR::inner_value_type;
+                array ar = test::make_array<scalar_value_type>(size);
+
+                CHECK(ar.owns_arrow_array());
+                CHECK(ar.owns_arrow_schema());
+            }
+
+            SUBCASE("not owning")
+            {
+                constexpr size_t offset = 0;
+                constexpr size_t size = 10;
+                using scalar_value_type = typename AR::inner_value_type;
+                ArrowSchema sc{};
+                ArrowArray ar{};
+                test::fill_schema_and_array<scalar_value_type>(sc, ar, size, offset, {});
+                array a(&ar, &sc);
+
+                CHECK(!a.owns_arrow_array());
+                CHECK(!a.owns_arrow_schema());
+            }
+        }
+        TEST_CASE_TEMPLATE_APPLY(owns_arrow_structure_id, testing_types);
+
+        TEST_CASE_TEMPLATE_DEFINE("get_arrow_structure", AR, get_arrow_structure_id)
+        {
+            constexpr size_t offset = 0;
+            constexpr size_t size = 10;
+            using scalar_value_type = typename AR::inner_value_type;
+            
+            ArrowSchema sc_ctrl{};
+            ArrowArray ar_ctrl{};
+            test::fill_schema_and_array<scalar_value_type>(sc_ctrl, ar_ctrl, size, offset, {});
+            auto pa_ctrl = primitive_array<scalar_value_type>(arrow_proxy(std::move(ar_ctrl), std::move(sc_ctrl)));
+
+            SUBCASE("not owning")
+            {
+                ArrowSchema sc{};
+                ArrowArray ar{};
+                test::fill_schema_and_array<scalar_value_type>(sc, ar, size, offset, {});
+                array a(&ar, &sc);
+
+                ArrowSchema* sc_ptr = nullptr;
+                ArrowArray* ar_ptr = nullptr;
+                a.get_arrow_array(ar_ptr).get_arrow_schema(sc_ptr);
+
+                auto pa = primitive_array<scalar_value_type>(arrow_proxy(ar_ptr, sc_ptr));
+
+                CHECK_EQ(pa, pa_ctrl);
+            }
+
+            SUBCASE("owning")
+            {
+                ArrowSchema sc{};
+                ArrowArray ar{};
+                test::fill_schema_and_array<scalar_value_type>(sc, ar, size, offset, {});
+                array a(std::move(ar), std::move(sc));
+
+                ArrowSchema* sc_ptr = nullptr;
+                ArrowArray* ar_ptr = nullptr;
+                a.get_arrow_array(ar_ptr).get_arrow_schema(sc_ptr);
+
+                auto pa = primitive_array<scalar_value_type>(arrow_proxy(ar_ptr, sc_ptr));
+
```

Comment:
You have to call the release callback of both array and schema and delete both pointers.

Reply:
`ar_ptr` points to `&ar` and `sc_ptr` points to `&sc` so they should not be deleted (`ar` and `sc` are allocated on the stack and will be destroyed at the end of the scope). The release methods of these structures must still be called though, I find it more intuitive to call them on the original structure (i.e. `ar` and `sc`).

Reply:
I mixed with extrat. You are right, a will release the array and schema.